site/src/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,163 @@
+import {
+  ReactElement,
+  ReactNode,
+  cloneElement,
+  createContext,
+  useContext,
+  useRef,
+  useState,
+} from "react";
+// This is used as base for the main Popover component
+// eslint-disable-next-line no-restricted-imports -- Read above
+import MuiPopover, {
+  type PopoverProps as MuiPopoverProps,
+} from "@mui/material/Popover";
+
+type TriggerMode = "hover" | "click";
+
+type TriggerRef = React.RefObject<HTMLElement>;
+
+type TriggerElement = ReactElement<{
+  onClick?: () => void;
+  ref: TriggerRef;
```

Comment:
Does this work with the intrinsic elements only (div, h1, etc.), or does it also work with any component that works with `forwardRef`?

I know Radix uses React's `ForwardRefExoticComponent` type under the hood, but I don't really know what it does, or if it's necessary here

Reply:
I know it works with any other elements like `<Button>` that supports `forwardRef`. `ForwardRefExoticComponent` looks dark magic 😆

Reply:
Yeah...it scares me, too